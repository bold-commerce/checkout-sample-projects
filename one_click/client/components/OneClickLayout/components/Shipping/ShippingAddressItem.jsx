import React from 'react';
import { RadioItem } from '../RadioItem';

const ShippingAddressItem = ({ address, onChange, selectedAddress, disabled }) => {
  const { address1, address2, city, province, zip, country, phone, first_name, last_name } = address;
  const addressArray = [address1, address2, city, province, zip, country];
  const addressCombined = addressArray.filter((addressString) => addressString !== '').join(', ');

  return (
    <li>
      <RadioItem 
        name="shipping_address"
        id={`address_${address.id}`}
        checked={address.id === selectedAddress}
        onChange={() => onChange(address)}
        disabled={disabled}
      >
        <p>{`${first_name} ${last_name}`}</p>
        <p>{addressCombined}</p>
        <p>{phone}</p>
      </RadioItem>
    </li>
  )
};

export default ShippingAddressItem;
