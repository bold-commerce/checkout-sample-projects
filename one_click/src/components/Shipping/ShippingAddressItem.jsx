import React from 'react';
import { RadioItem } from '../RadioItem';

const ShippingAddressItem = ({ address, onChange, selectedAddress, disabled }) => {
  const { address_line_1, address_line_2, city, province, postal_code, country, phone_number, first_name, last_name } = address;
  const addressArray = [address_line_1, address_line_2, city, province, postal_code, country];
  const addressCombined = addressArray.filter((addressString) => addressString !== '').join(', ');

  return (
    <li>
      <RadioItem 
        name="shipping_address"
        id={`address_${address.id}`}
        checked={address.id === selectedAddress}
        onChange={() => onChange(address)}
        disabled={disabled}
      >
        <p>{`${first_name} ${last_name}`}</p>
        <p>{addressCombined}</p>
        <p>{phone_number}</p>
      </RadioItem>
    </li>
  )
};

export default ShippingAddressItem;
