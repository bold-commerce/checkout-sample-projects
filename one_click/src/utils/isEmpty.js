const isEmpty = (obj) => {
  return JSON.stringify(obj) === JSON.stringify({});
};

export default isEmpty;