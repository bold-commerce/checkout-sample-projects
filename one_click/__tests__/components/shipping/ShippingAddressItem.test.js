import React from 'react';
import ShippingAddressItem from '../../../src/components/Shipping/ShippingAddressItem';
import { render } from '@testing-library/react';
import { exampleAddress } from '../../utils/addressHelpers';

describe('ShippingAddressItem', () => {
    test('renders ShippinAddressItem component', () => {
        const { asFragment } = render(
            <ShippingAddressItem
                address={exampleAddress}
                onChange={null}
                selectedAddress={null}
                disabled={false}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders ShippinAddressItem component with onChange', () => {
        const { asFragment } = render(
            <ShippingAddressItem
                address={exampleAddress}
                onChange={() => {}}
                selectedAddress={null}
                disabled={false}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders ShippinAddressItem component disabled', () => {
        const { asFragment } = render(
            <ShippingAddressItem
                address={exampleAddress}
                onChange={null}
                selectedAddress={null}
                disabled
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});