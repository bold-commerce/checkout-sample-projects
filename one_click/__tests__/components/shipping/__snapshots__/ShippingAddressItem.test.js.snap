// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingAddressItem renders ShippinAddressItem component 1`] = `
<DocumentFragment>
  <li>
    <div
      class="RadioItem"
    >
      <input
        id="address_undefined"
        name="shipping_address"
        type="radio"
      />
      <label
        class="RadioItem__Content"
        for="address_undefined"
      >
        <p>
          John Doe
        </p>
        <p>
          , , Winnipeg, Manitoba, , Canada
        </p>
        <p />
      </label>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ShippingAddressItem renders ShippinAddressItem component disabled 1`] = `
<DocumentFragment>
  <li>
    <div
      class="RadioItem"
    >
      <input
        disabled=""
        id="address_undefined"
        name="shipping_address"
        type="radio"
      />
      <label
        class="RadioItem__Content"
        for="address_undefined"
      >
        <p>
          John Doe
        </p>
        <p>
          , , Winnipeg, Manitoba, , Canada
        </p>
        <p />
      </label>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ShippingAddressItem renders ShippinAddressItem component with onChange 1`] = `
<DocumentFragment>
  <li>
    <div
      class="RadioItem"
    >
      <input
        id="address_undefined"
        name="shipping_address"
        type="radio"
      />
      <label
        class="RadioItem__Content"
        for="address_undefined"
      >
        <p>
          John Doe
        </p>
        <p>
          , , Winnipeg, Manitoba, , Canada
        </p>
        <p />
      </label>
    </div>
  </li>
</DocumentFragment>
`;
